core/src/test/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequestTests.java

```diff
@@ -94,4 +102,92 @@ public void testPutMappingRequestSerialization() throws IOException {
             }
         }
     }
+
+    public void testToXContent() throws IOException {
+        PutMappingRequest request = new PutMappingRequest("foo");
+
+        request.type("my_type");
+
+        XContentBuilder mapping = JsonXContent.contentBuilder().startObject();
+        mapping.startObject("properties");
+        mapping.startObject("email");
+        mapping.field("type", "text");
+        mapping.endObject();
+        mapping.endObject();
+        mapping.endObject();
+        request.source(mapping);
+
+        String actualRequestBody = Strings.toString(request);
+
+        String expectedRequestBody = "{\"properties\":{\"email\":{\"type\":\"text\"}}}";
+
+        assertEquals(expectedRequestBody, actualRequestBody);
+    }
+
+    public void testToAndFromXContent() throws IOException {
+
+        final PutMappingRequest putMappingRequest = createTestItem();
+
+        boolean humanReadable = randomBoolean();
+        final XContentType xContentType = randomFrom(XContentType.values());
+        BytesReference originalBytes = toShuffledXContent(putMappingRequest, xContentType, EMPTY_PARAMS, humanReadable);
+
+        PutMappingRequest parsedPutMappingRequest = new PutMappingRequest();
+        parsedPutMappingRequest.source(originalBytes, xContentType);
+
+        assertMappingsEqual(putMappingRequest.source(), parsedPutMappingRequest.source());
+    }
+
+    private void assertMappingsEqual(String expected, String actual) throws IOException {
+
+        XContentParser expectedJson = createParser(XContentType.JSON.xContent(), expected);
+        XContentParser actualJson = createParser(XContentType.JSON.xContent(), actual);
+        assertEquals(expectedJson.mapOrdered(), actualJson.mapOrdered());
+    }
+
+    /**
+     * Returns a random {@link PutMappingRequest}.
+     */
+    private static PutMappingRequest createTestItem() throws IOException {
+        String index = randomAlphaOfLength(5);
+
+        PutMappingRequest request = new PutMappingRequest(index);
+
+        String type = randomAlphaOfLength(5);
+        request.type(type);
+        request.source(randomMapping());
+
+        return request;
+    }
+
+    private static XContentBuilder randomMapping() throws IOException {
```

Comment:
@javanna - One more question: these two methods are duplicated in `CreateIndexRequestTests`; what would be a sensible place to commonise them?